### ISSUE TITLE
Invalid symlinks in cache directory on symlinked home

### ISSUE THREAD

**Issue by b-x** [pex-tool/pex#2572]
Steps to reproduce:
```
mkdir -p /tmp/a/b/c
ln -s /tmp/a/b/c /tmp/lnk
HOME=/tmp/lnk  my_pex_app
```
It gives following error:
```
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/tmp/lnk/.cache/pex/unzipped_pexes/0/3d53a34d47e626490f86d2f53f2a547241faa8fd/__main__.py", line 227, in <module>
    result, should_exit, is_globals = boot(
                                      ^^^^^
  File "/tmp/lnk/.cache/pex/unzipped_pexes/0/3d53a34d47e626490f86d2f53f2a547241faa8fd/__main__.py", line 216, in boot
    from pex.globals import Globals
ModuleNotFoundError: No module named 'pex'
```
And the probable reason is broken symlink:
```
$ file /tmp/lnk/.cache/pex/unzipped_pexes/0/3d53a34d47e626490f86d2f53f2a547241faa8fd/__pex__
/tmp/lnk/.cache/pex/unzipped_pexes/0/3d53a34d47e626490f86d2f53f2a547241faa8fd/__pex__: broken symbolic link to ../../../../../../a/b/c/.cache/pex/user_code/0/e71a56f11399e5f48fb037e733eb39e4123b8575/__pex__
```

**Comment by jsirois** [comment]
Thanks for the report @b-x. I should have a fix for this included in today's 2.20.4 bug fix release.

**Comment by jsirois** [comment]
There was one case not covered by #2574 that @benjyw fixed in #2626.
